tests/unit/browser/test_commands.py

```diff
@@ -0,0 +1,212 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2018 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test TabbedBrowser command dispatche."""
+
+from unittest import mock
+
+import pytest
+
+from qutebrowser.browser import commands
+from qutebrowser.completion.models import completionmodel
+from qutebrowser.commands import cmdexc
+from qutebrowser.mainwindow import tabbedbrowser, tabwidget
+from qutebrowser.utils import objreg
+
+
+class TestBuffer:
+    """Test logic for :buffer in CommandDispatcher."""
+
+    @pytest.fixture
+    def tabbed_browser(self):
+        """Simple tabbed browser mock that doesn't require winreg."""
+        tabbed_browser = mock.create_autospec(tabbedbrowser.TabbedBrowser)
+        tabbed_browser.widget = mock.create_autospec(tabwidget.TabWidget)
+        yield tabbed_browser
+
+    @pytest.fixture
+    def app(self, stubs):
+        """Simple tabbed browser mock that doesn't require winreg."""
+        app = stubs.FakeQApplication()
+        objreg.register('app', app)
+        yield app
+        objreg.delete('app')
+
+    @mock.patch.object(commands.CommandDispatcher, 'openurl')
```

Comment:
Please use the [mocker fixture](https://github.com/pytest-dev/pytest-mock) from pytest-mock instead - mixing pytest fixtures and decorators comes with some weird caveats.

(same for other tests below, too)